build.gradle

```diff
@@ -41,6 +42,28 @@ distributions {
     }
 }
 
+jar {
+    manifest {
+        attributes ("Main-Class" : "io.logz.jmx2graphite.Jmx2GraphiteJolokia",
+                    "Premain-Class": "io.logz.jmx2graphite.Jmx2GraphiteJavaAgent")
+    }
+}
+
+task javaAgent(dependsOn: [shadowJar]) {
+}
+
+shadowJar {
+    manifest.from jar.manifest
+    classifier = 'javaagent'
+    relocate 'org.apache', 'io.logz.org.apache'
+    relocate 'com.fasterxml.jackson', 'io.logz.com.fasterxml.jackson'
+    relocate 'io.dropwizard.metrics', 'io.logz.ch.io.dropwizard.metrics'
+    relocate 'com.typesafe', 'io.logz.com.typesafe'
```

Comment:
In Maven Shade plugin you can specifying include using wildcards.
If that's not possible here, you can simply use Gradle powers and iterate over the dependencies.

There's a good chance you'll forget to shadow a lib in the future. If it's not too much work (in Maven it's not) it's worth the effort.